src/rust/ide/src/view/layout.rs

```diff
@@ -30,11 +31,12 @@ shared! { ViewLayout
 /// Initial implementation of ViewLayout with a TextEditor and NodeEditor.
 #[derive(Debug)]
 pub struct ViewLayoutData {
-    text_editor   : TextEditor,
-    node_editor   : NodeEditor,
-    node_searcher : NodeSearcher,
-    size          : Vector2<f32>,
-    logger        : Logger
+    text_editor    : TextEditor,
+    node_editor    : NodeEditor,
+    node_searcher  : NodeSearcher,
+    size           : Vector2<f32>,
+    logger         : Logger,
+    mouse_position : Uniform<Vector2<i32>>
```

Comment:
Dont use uniforms. Uniforms are shader things! Using shader things in controller logic is very bad design. Use `scene::Mouse.frp` instead.